### PR TITLE
Merge 2.3 to main

### PR DESCRIPTION
**Pull request by treid314**
#### What this PR does
As part of the 2.3 release we want to merge the release branch to main to update the changelog and meta files.

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -95,15 +87,21 @@
 * [CHANGE] Purger: removed the purger component and moved its API endpoints `/purger/delete_tenant` and `/purger/delete_tenant_status` to the compactor at `/compactor/delete_tenant` and `/compactor/delete_tenant_status`. The new endpoints on the compactor are stable. #2644
 * [CHANGE] Memberlist: Change the leave timeout duration (`-memberlist.leave-timeout duration`) from 5s to 20s and connection timeout (`-memberlist.packet-dial-timeout`) from 5s to 2s. This makes leave timeout 10x the connection timeout, so that we can communicate the leave to at least 1 node, if the first 9 we try to contact times out. #2669
 * [CHANGE] Alertmanager: return status code `412 Precondition Failed` and log info message when alertmanager isn't configured for a tenant. #2635
-* [CHANGE] Distributor: if forwarding rules are used to forward samples, exemplars are now removed from the request. #2710
+* [CHANGE] Distributor: if forwarding rules are used to forward samples, exemplars are now removed from the request. #2710, #2725
```
Comment (pstibrany) [inline CHANGELOG.md:90]: We don't use comma between PR numbers.

If we're adding it to 2.3.0 release, we should remove entry for PR 2725 from `main` section.

Comment (treid314) [reply CHANGELOG.md]: Updated

**Review by johannaratliff — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -135,21 +113,33 @@
 * [BUGFIX] Fix reporting of tracing spans from PromQL engine. #2707
 * [BUGFIX] Apply relabel and drop_label rules before forwarding rules in the distributor. #2703
 * [BUGFIX] Distributor: Register `cortex_discarded_requests_total` metric, which previously was not registered and therefore not exported. #2712
+* [BUGFIX] Ruler: fix not restoring alerts' state at startup. #2648
+* [BUGFIX] Ingester: Fix disk filling up after restarting ingesters with out-of-order support disabled while it was enabled before. #2799
+* [BUGFIX] Memberlist: retry joining memberlist cluster on startup when no nodes are resolved. #2837
+* [BUGFIX] Query-frontend: fix incorrect mapping of http status codes 413 to 500 when request is too large. #2819
+* [BUGFIX] Alertmanager: revert upstream alertmananger to v0.24.0 to fix panic when unmarshalling email headers #2924 #2925
+* [BUGFIX] Fix sanity check done on configured filesystem directories when running Alertmanager in microservices mode. #2947
```
Comment (pstibrany) [inline CHANGELOG.md:121]: This entry should be added to:

> * [ENHANCEMENT] Add sanity check at startup to ensure the configured filesystem directories don't overlap for different components. #2828

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,78 +4,42 @@
...
-* [CHANGE] Compactor: `-compactor.partial-block-deletion-delay` must either be set to 0 (to disable partial blocks deletion) or a value higher than `4h`. #2787
-* [CHANGE] Query-frontend: CLI flag `-query-frontend.align-querier-with-step` has been deprecated. Please use `-query-frontend.align-queries-with-step` instead. #2840
 * [CHANGE] Distributor: change the default value of `-distributor.remote-timeout` to `2s` from `20s` and `-distributor.forwarding.request-timeout` to `2s` from `10s` to improve distributor resource usage when ingesters crash. #2728
-* [CHANGE] Ingester: changed default setting for `-ingester.ring.readiness-check-ring-health` from `true` to `false`. #2953
 * [CHANGE] Anonymous usage statistics tracking: added the `-ingester.ring.store` value. #2981
-* [FEATURE] Introduced an anonymous usage statistics tracking (enabled by default), to help Mimir maintainers make better decisions to support the open source community. The tracking system anonymously collects non-sensitive, non-personally identifiable information about the running Mimir cluster. #2643 #2662 #2685 #2732 #2733 #2735 #2939 #2940
```
Comment (pstibrany) [inline CHANGELOG.md:15]: This entry needs to be split between 2.3.0 and `main`. PRs #2939 #2940 are only in `main`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,78 +4,42 @@
...
-* [CHANGE] Ingester: removed deprecated `-blocks-storage.tsdb.isolation-enabled` option. TSDB-level isolation is now always disabled in Mimir. #2782
-* [CHANGE] Compactor: `-compactor.partial-block-deletion-delay` must either be set to 0 (to disable partial blocks deletion) or a value higher than `4h`. #2787
-* [CHANGE] Query-frontend: CLI flag `-query-frontend.align-querier-with-step` has been deprecated. Please use `-query-frontend.align-queries-with-step` instead. #2840
 * [CHANGE] Distributor: change the default value of `-distributor.remote-timeout` to `2s` from `20s` and `-distributor.forwarding.request-timeout` to `2s` from `10s` to improve distributor resource usage when ingesters crash. #2728
-* [CHANGE] Ingester: changed default setting for `-ingester.ring.readiness-check-ring-health` from `true` to `false`. #2953
```
Comment (pstibrany) [inline CHANGELOG.md:13]: I can't find this changelog entry in the merged changelog.

File: CHANGELOG.md (hunk 2 above)
Comment (treid314) [reply CHANGELOG.md:121]: Is there a process to finding these? Or do you just know? I think I'm missing a step here.

File: CHANGELOG.md (hunk 3 above)
Comment (pracucci) [reply CHANGELOG.md:15]: In `main` changelog we could write:

```
* [CHANGE] Anonymous usage statistics tracking has been enabled by default, to help Mimir maintainers make better decisions to support the open source community. #2939
* [CHANGE] Anonymous usage statistics tracking: added the minimum and maximum value of `-ingester.out-of-order-time-window`. #2940
```

File: CHANGELOG.md (hunk 4 above)
Comment (pracucci) [reply CHANGELOG.md:13]: This is not part of 2.3. It should stay in `main`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,78 +4,42 @@
...
-* [BUGFIX] Ruler: fix panic when `ruler.external_url` is explicitly set to an empty string (`""`) in YAML. #2915
-* [BUGFIX] Fix sanity check done on configured filesystem directories when running Alertmanager in microservices mode. #2947
 * [BUGFIX] Querier: Fix 400 response while handling streaming remote read. #2963
 * [BUGFIX] Fix a bug causing query-frontend, query-scheduler, and querier not failing if one of their internal components fail. #2978
 
 ### Mixin
 
-* [CHANGE] Dashboards: remove the "Cache - Latency (old)" panel from the "Mimir / Queries" dashboard. #2796
 * [CHANGE] Alerts: MimirQuerierAutoscalerNotActive is now critical and fires after 1h instead of 15m. #2958
 * [FEATURE] Dashboards: added support to experimental read-write deployment mode. #2780
```
Comment (pracucci) [inline CHANGELOG.md:22]: This should be removed because included in the 2.3 changelog.

File: CHANGELOG.md (hunk 2 above)
Comment (treid314) [reply CHANGELOG.md:121]: https://github.com/grafana/mimir/pull/2995/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR98 That's added here

Comment (treid314) [reply CHANGELOG.md:121]: Oh I see

File: CHANGELOG.md (hunk 3 above)
Comment (pracucci) [reply CHANGELOG.md:15]: This entry has been lost. You should move it to 2.3 changelog, removing `#2939 #2940`.

**Review by pracucci — APPROVED** [review]
LGTM! If there's anything missing, we'll fix in a follow up PR. Thanks!

**Comment by pstibrany** [comment]
Please make sure to merge this with a merge commit, not Squash and Merge. Thank you!

**Pull request merged**